src/fsharp/TypeChecker.fs

```diff
@@ -10476,7 +10477,7 @@ and TcMethodArg cenv env (lambdaPropagationInfo, tpenv) (lambdaPropagationInfoFo
                   if AddCxTypeMustSubsumeTypeMatchingOnlyUndoIfFailed env.DisplayEnv cenv.css mArg adjustedCalledTy argTy then
                      yield info |]
 
-    CallerArg(argTy, mArg, isOpt, e'), (lambdaPropagationInfo, tpenv)
+    { Type = argTy; Range = mArg; IsOptional = isOpt; Expr = e' }, (lambdaPropagationInfo, tpenv)
```

Comment:
Could you undo this change to CallerArg pleae?  It's cosmetic and isn't essential to your improvement, thanks!!

Reply:
@dsyme, done!

I want to bring attention to few points though:

* I initially did the change because in the debugger variable inspector it shows everything twice
* the fields were unammed (they now are) so you'd see Item1 etc, not great when you aren't familiar with the code and start looking into it
* you can't do an exhaustive find usages on fields, with record you always can
* when properties are defined, two ways to reach the same data in code
* minor, but field access vs property access 

What is the right approach to bring such improvement (affects documentation, debuggability, approachability, searchability) in the compiler codebase?